stable/aws-cloudwatch-metrics/templates/daemonset.yaml

```diff
@@ -69,6 +72,14 @@ spec:
       - name: varlibdocker
         hostPath:
           path: /var/lib/docker
+      - name: containerdsock
+        hostPath:
+        {{- if .Values.bottlerocket }}
```

Comment:
I agree that this would solve the immediate problem at hand. However, this value name of `bottlerocket` is a one-way door for a couple reasons:

1) What happens if Bottlerocket decides to change their containerd sock path again?
2) What happens if there is another host OS we want to target in the future that has a different sock path?

For this reason, I think it would be better to have a `containerdSockPath` value name and make it a string, defaulted to `/run/containerd/containerd.sock` with a comment informing the user to set this to `/run/dockershim.sock` when using Bottlerrocket as a host OS.

Reply:
bottlerocket rejoined the fold in 1.23 and is planning to backport

https://github.com/bottlerocket-os/bottlerocket/issues/2212